src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -266,147 +232,94 @@ public override Task CloseAsync(WebSocketCloseStatus closeStatus, string? status
             {
                 throw new WebSocketException(WebSocketError.InvalidState, SR.Format(SR.net_WebSockets_InvalidState, fastState, "Connecting, Open, CloseSent, Aborted"));
             }
-
-#if FEATURE_WASM_THREADS
-            return _innerWebSocket!.SynchronizationContext.Post(() =>
+            WebSocketState state;
+            lock (_lockObject)
             {
-                Task promise;
-                lock (_thisLock)
+                state = State;
+                if (state == WebSocketState.Closed)
                 {
-                    ThrowIfDisposed();
-#endif
-                    var state = State;
-                    if (state == WebSocketState.None || state == WebSocketState.Closed)
-                    {
-                        throw new WebSocketException(WebSocketError.InvalidState, SR.Format(SR.net_WebSockets_InvalidState, state, "Connecting, Open, CloseSent, Aborted"));
-                    }
-                    if (state != WebSocketState.Open && state != WebSocketState.Connecting && state != WebSocketState.Aborted && state != WebSocketState.CloseSent)
-                    {
-                        return Task.CompletedTask;
-                    }
+                    return Task.CompletedTask;
+                }
+                if (state == WebSocketState.None || state == WebSocketState.Closed)
+                {
+                    throw new WebSocketException(WebSocketError.InvalidState, SR.Format(SR.net_WebSockets_InvalidState, state, "Connecting, Open, CloseSent, Aborted"));
+                }
 
-#if FEATURE_WASM_THREADS
-                    promise = CloseAsyncCore(closeStatus, statusDescription, state != WebSocketState.Aborted, cancellationToken);
-                } //lock will unlock synchronously before task is resolved!
-                return promise;
-            });
-#else
-            return CloseAsyncCore(closeStatus, statusDescription, state != WebSocketState.Aborted, cancellationToken);
-#endif
+            } // lock
+            return CloseAsyncCore(closeStatus, statusDescription, state != WebSocketState.Aborted, cancellationToken, fastState);
         }
 
         public override void Abort()
         {
-#if FEATURE_WASM_THREADS
-            if (_disposed)
+            WebSocketState state;
+            lock (_lockObject)
             {
-                return;
-            }
-            _innerWebSocket?.SynchronizationContext.Send(static (BrowserWebSocket self) =>
-            {
-                lock (self._thisLock)
+                if (_disposed)
                 {
-                    AbortCore(self, self.State);
+                    return;
                 }
-            }, this);
-#else
-            AbortCore(this, this.State);
-#endif
+                state = State;
+            } // lock
+            AbortCore(this, state);
```

Comment:
is it intentional that we no longer abort within the lock?

Reply:
refactored